### DISCUSSION TITLE
Utilizzo dei regex nei vari repository

### DISCUSSION THREAD

**Discussion by VaiTon** [csunibo/csunibo.github.io#13] [Domande]
L'idea è quella di utilizzare workflow condivisi per tutti i repository, in modo da diminuire il lavoro per aggiornare versioni / cambiare workflow. 

In particolare, questa discussione si concentra sul controllo dei nomi dei file.

Al momento, il controllo viene fatto tramite una GitHub Action custom (https://github.com/csunibo/filenames-linter), ma si vorrebbe utilizzare [csurename](https://github.com/csunibo/csurename) in the long run.

Dalla discussione su Telegram la proposta che sembra più sensata / attuabile è:
> Utilizzare un unica serie di RegEx, eventualmente sovrascrivibile da un file di configurazione **nel repository** (eg. `.csconfig.json`).

Il problema adesso è quindi capire in che modo applicare questo regex, se su ogni file, ogni cartella, alcune cartelle si alcune no, etc...

Qual è la vostra idea a riguardo?

**Comment by lucat1** [comment]
Attualmente le regexp sono scomode per quelle prove dove bisogna caricare file con nomenclature (dettate dal linguaggio di programmazione o altri strumenti specifici) che non si adeguano al nostro utilizzo del kebab-case. Spesso questi tipi di file sono più di uno per prova d'esame, quindi vengono inseriti in una cartella che (attualmente) viene esplorata ricorsivamente e le regole del kebab-casing sono applicate su tutto.

L'idea di @foxyseta (che appoggio) è di cambiare l'attuale politica in:
- se analizziamo un file deve rispettare la regexp
- se analizziamo una cartella ho due casi:
  1. rispetta la regexp. Allora non la esploro ricorsivamente (a differenza di quanto viene fatto ora)
  2. se non rispetta la regexp vado di ricorsione.

Questo ci consente di usare le nomenclature necessarie per lo specifico caso, lasciando comunque un certo livello di ordine fino a quando non si entra a guardare una soluzione specifica dentro una cartella.

**Reply by foxyseta** [reply]
Proposta di modifica: se la cartella non rispetta pa regexp oltre a esplorarla ricorsivamente imponiamo anche che almeno sia in kebab-case.

**Reply by lucat1** [reply]
Si assolutamente, evita un po' di casino e non vedo che problemi potrebbe dare.

**Reply by VaiTon** [reply]
Per me va bene, se decidiamo su questa possiamo chiudere e implementarla.

**Reply by lucat1** [reply]
Sentiamo il feedback degli altri:
@csunibo/informatica-1 @csunibo/informatica-2 @csunibo/informatica-3 (ingegneria non so chi taggare)

**Reply by kocierik** [reply]
io sono d'accordo

**Reply by lucat1** [reply]
Ci lavoriamo asap, assieme all'integrazione con synta

**Comment by foxyseta** [comment]
(Flussi di lavoro e sintassi sinta condivisi in `config`)
